sql/src/main/java/com/rilldata/calcite/models/ArtifactStore.java

```diff
@@ -2,31 +2,42 @@
 
 import java.util.HashMap;
 import java.util.HashSet;
+import java.util.List;
 import java.util.Map;
 import java.util.Set;
 
-public class InMemoryArtifactManager implements ArtifactManager
+public class ArtifactStore
 {
   public static final Set<String> EMPTY_SET = new HashSet<>(0);
   Map<ArtifactType, Map<String, Artifact>> artifacts = new HashMap<>();
 
-  @Override public void saveArtifact(Artifact artifact)
+  public ArtifactStore() {}
+
+  public ArtifactStore(List<Artifact> artifacts)
+  {
+    for (Artifact artifact : artifacts) {
+      saveArtifact(artifact);
+    }
+  }
+
+  public void saveArtifact(Artifact artifact)
   {
     artifacts.computeIfAbsent(artifact.getType(), type -> new HashMap<>());
     Artifact existing = artifacts.get(artifact.getType()).putIfAbsent(artifact.getName().toLowerCase(), artifact);
     if (existing != null) {
-      System.out.printf("Artifact with name %s of type %s already exists, overwriting!%n", artifact.getName().toLowerCase(),
+      System.out.printf("Artifact with name %s of type %s already exists, overwriting!\n",
+          artifact.getName().toLowerCase(),
```

Comment:
Will this lead to logs when validating a metrics view that already exists? That will happen when people want to update metrics views. Product isn't happy about unexpected logs during normal usage

Reply:
removed this in `runtime-metrics` PR